### PR TITLE
HEC-789: save tax check endpoint

### PR DESCRIPTION
**Pull request by AndyHWChung**

### CODE REVIEW

**Review by priyankadhiman3006 — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/hec/services/TaxCheckCodeGeneratorService.scala [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  private def randomElement[A](l: List[A]): A = l(Random.nextInt(l.length))
+
+  private val allowedChars: List[Char] = {
+    val allowedLetters = ('A' to 'Z').toList.diff(List('I', 'O', 'S', 'U', 'V', 'W'))
+    val allowedDigits  = ('0' to '9').toList.diff(List('0', '1', '5'))
+    allowedLetters ::: allowedDigits
+  }
+
+  def next(): HECTaxCheckCode = {
```
[the remark this answers is not in the document]
Comment (AndyHWChung) [reply app/uk/gov/hmrc/hec/services/TaxCheckCodeGeneratorService.scala:42]: yeah, that makes sense - I'll change it

File: app/uk/gov/hmrc/hec/models/licence/LicenceDetails.scala [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.hec.models.licence
+
+import play.api.libs.json.{Json, OFormat}
+
+final case class LicenceDetails(
```
[the remark this answers is not in the document]
Comment (AndyHWChung) [reply app/uk/gov/hmrc/hec/models/licence/LicenceDetails.scala:21]: whoops, i meant to put tax situation under `TaxDetails` -  I'll stick it in now

File: app/uk/gov/hmrc/hec/services/TaxCheckCodeGeneratorService.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (AndyHWChung) [reply app/uk/gov/hmrc/hec/services/TaxCheckCodeGeneratorService.scala]: updated

File: app/uk/gov/hmrc/hec/models/licence/LicenceDetails.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (AndyHWChung) [reply app/uk/gov/hmrc/hec/models/licence/LicenceDetails.scala:21]: updated - FE is updated as well now

**Review by priyankadhiman3006 — APPROVED** [review]

**Pull request merged**